packages/visualizations/src/components/Table/Row.svelte

```diff
@@ -0,0 +1,83 @@
+<script lang="ts">
+    import Cell from './Cell';
+    import ZoomIcon from './ZoomIcon.svelte';
+    import type { Column, Rows } from './types';
+
+    export let columns: Column[];
+    export let rows: Rows | undefined;
+    export let record: Record<string, unknown>;
+    let isRowHovered = false;
+
+    $: onMouseEnter = () => {
+        if (rows?.onMouseEnter) {
+            rows.onMouseEnter(record);
+        }
+        isRowHovered = true;
+    };
+
+    $: onMouseLeave = () => {
+        if (rows?.onMouseLeave) {
+            rows.onMouseLeave(record);
+        }
+        isRowHovered = false;
+    };
+
+    $: onClick = () => {
+        if (rows?.onClick) {
+            rows.onClick(record);
+        }
+    };
+</script>
+
+<tr on:mouseenter={onMouseEnter} on:mouseleave={onMouseLeave}>
```

Comment:
It's a micro optimization, but ideally we shouldn't have these handlers attached at all if there's no `rows?.onClick`.

Reply:
Now the with the latest changes, even if `rowProps` is defined, we can't be sure that `onMouseEnter` is defined.

Complete check should be `rowProps?.onMouseEnter`.
But do we want do to it like this as `handleMouseEnter` is already making the check on `onMouseEnter`? 


Can we do something like this? 

```svelte
    $: handleMouseEnter = onMouseEnter ?  () => {
        onMouseEnter(record);
        isRowHovered = true;
    } : undefined;
    
    <tr on:mouseenter={handleMouseEnter} ... />
 ```